From 6ac4929e0bb77509eab2a7750a2fa3dcc85328b0 Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Wed, 27 Jul 2022 03:55:41 -0400
Subject: [PATCH 3610/4585] [Huawei] efi/libstub: arm64: support for multiple
 memmap parameters

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187251

--------------------------------

memmap=nn[KMG]$ss[KMG]
Mark specific memory as reserved. multiple regions of memory
can be reserved.
Example:
        memmap=64K$0x18690000,0x10000$19690000
        or
        memmap=64K$0x18690000 memmap=0x10000$19690000

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
---
 drivers/firmware/efi/libstub/arm64-stub.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/firmware/efi/libstub/arm64-stub.c b/drivers/firmware/efi/libstub/arm64-stub.c
index 01016d5a2fc9..251623f97948 100644
--- a/drivers/firmware/efi/libstub/arm64-stub.c
+++ b/drivers/firmware/efi/libstub/arm64-stub.c
@@ -50,7 +50,8 @@ efi_parse_memmap(char *p, unsigned long long *start, unsigned long long *size)
 
 void efi_parse_option_memmap(const char *str)
 {
-	int idx = 0, rc;
+	int rc;
+	static int idx;
 	char *k, *p = (char *)str;
 
 	while (p && (idx < MAX_MEMMAP_REGIONS)) {
@@ -60,7 +61,7 @@ void efi_parse_option_memmap(const char *str)
 
 		rc = efi_parse_memmap(p, &mem_avoid[idx].start, &mem_avoid[idx].size);
 		if (rc < 0)
-			efi_err("Failed to parse memmap cmdlines, index: %d\n", idx);
+			efi_err("Failed to parse memmap cmdlines, index: %d, str: %s\n", idx, p);
 
 		p = k;
 		idx++;
-- 
2.17.1

